### ISSUE TITLE
The Nesting Rule

### ISSUE THREAD

**Issue by kmikael** [realm/SwiftLint#190]
What is the nesting rules supposed to do? I think it doesn't work.

For example, with the nesting rule enabled, I get a warning, `Nesting Violation: Types should be nested at most 1 level deep (nesting)`:

```swift
class View: UIView {
    enum State {
        case Ready, Executing, Finished
    }
}
```

But the nesting is correct.

**Comment by jpsim** [comment]
The nesting rule will throw a violation when types are nested more than 1 level, or statements more than 5 levels. For example:

**OK**
```swift
class Class0 {
    class Class1 {
    }
}
```
```
Done linting! Found 0 violations, 0 serious in 1 file.
```

**Style Violation**
```swift
class Class0 {
    class Class1 {
        class Class2 {
        }
    }
}
```
```
file.swift:3:9: warning: Nesting Violation: Types should be nested at most 1 level deep (nesting)
Done linting! Found 1 violation, 0 serious in 1 file.
```

What you're found is a bug which treated enum cases as a type declaration, which meant that enum *cases* could also only be nested one level deep, which is not the intention. I just fixed in 2131613166b12677a09c6b2d1b48d9ce4cb7093f. Thanks for reporting this!

**Comment by eaigner** [comment]
I still have this issue

**Comment by jpsim** [comment]
@eaigner based on this and the several other bug reports you've filed, you need to update your version of SwiftLint version to the latest (0.9.1), which has all these features.

You can tell which version of SwiftLint you're using by running `swiftlint version`.

**Comment by eaigner** [comment]
Yep. Totally my fault. Sorry. (was `0.3.0`)

**Comment by AlexIzh** [comment]
Can I change settings for this rule? I mean, what if I want to have this rule but I want to have level for type = 3 or something like this

**Comment by jpsim** [comment]
You can't change the thresholds for `nesting_rule` at the moment, but the rule could easily be modified to allow such configuration.

**Comment by erenkabakci** [comment]
The issue still persists on `0.18.1`

I get the warning with following setup
```
struct A {
  func Foo {
    enum Boo {
      // Warning is caused because of this
  }
}
```

Is it because of the enum inside a func in a struct, or am I missing out something here.

**Comment by sureshxaton** [comment]
I am using 0.24.0 and still facing this
 struct S {
        enum Value: String { //Warning caused
            case a, b
        }
    }

**Comment by ornithocoder** [comment]
@sbrighiu try this (assume you want `3` for nesting)

```
  nesting:
    type_level: 3
```
